core/src/wallet/ripple/RippleLikeAccount.cpp

```diff
@@ -123,9 +123,18 @@ namespace ledger {
             operation.trust = std::make_shared<TrustIndicator>();
             operation.date = transaction.receivedAt;
 
+            auto setAmount = [&](Operation& operation) {
+                if (transaction.status == "tesSUCCESS") {
+                    operation.amount = transaction.value;
+                } else {
+                    // if the status of the transaction is not correct, we set the operation’s amount to
+                    // zero as it’s failed (yet fees were still paid)
+                    operation.amount = BigInt::ZERO;
+                }
+            };
```

Comment:
Just a shortcut to prevent having to duplicate that logic at the two calling sites below. Yes, it could be a plain function. Do you think it’d be clearer to change that?

Reply:
I think we have to use lambda only when we want to pass a function as function parameter OR when the captured-variables are necessary. 
In my opinion the lambda syntax adds cognitive complexity so if we can scope that logic into a old-plained function with a good name it's better.

Reply:
Case in point. I’ll change it, thanks!